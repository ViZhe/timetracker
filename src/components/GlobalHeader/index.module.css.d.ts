
export const header: string;
export const trigger: string;
