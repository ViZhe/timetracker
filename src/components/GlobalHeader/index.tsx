
import React, { Component } from 'react';
import { Layout, Icon } from 'antd';

import styles from './index.module.css';


type Props = {
  collapsed: boolean;
  onClickTrigger: () => void;
};

const { Header } = Layout;

class GlobalHeader extends Component<Props> {
  render() {
    const { collapsed, onClickTrigger } = this.props;
    return (
      <Header className={styles.header}>
        <Icon
          className={styles.trigger}
          type={collapsed ? 'menu-unfold' : 'menu-fold'}
          onClick={onClickTrigger}
        />
      </Header>
    );
  }
}


export default GlobalHeader;
