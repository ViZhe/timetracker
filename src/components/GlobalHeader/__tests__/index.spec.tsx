
import { Icon } from 'antd';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import GlobalHeader from '../../GlobalHeader';


describe('<GlobalHeader />', () => {
  let component:ShallowWrapper;
  const collapsed = false;
  const onClick = jest.fn();

  beforeEach(() => {
    component = shallow(
      <GlobalHeader
        collapsed={collapsed}
        onClickTrigger={onClick}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should be collapsed by default', () => {
    expect(component.find(Icon).get(0).props.type).toEqual('menu-fold');
  });

  it('should call onClick once', () => {
    component.find(Icon).simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
