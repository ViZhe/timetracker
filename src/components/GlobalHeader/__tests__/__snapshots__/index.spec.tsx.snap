// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalHeader /> should match snapshot 1`] = `
<Adapter
  className="header"
>
  <Icon
    className="trigger"
    onClick={[MockFunction]}
    type="menu-fold"
  />
</Adapter>
`;
